### DISCUSSION TITLE
[BE] 그룹의 현재 상태를 어떻게 할까?

### DISCUSSION THREAD

**Discussion by nbalance97** [woowacourse-teams/2022-momo#44] [General]
## 문제점
모임 도메인에서 현재 모임이 열린 모임인지, 닫힌 모임인지의 상태 관리의 책임을 어디서 할지에 대한 뜨거운 논쟁

## 의견
1안. 도메인에서 상태를 갖는다

2안. 로직으로써 상태를 관리한다

**Comment by usageness** [comment]
2안 찬성합니당 

- 도메인에서 상태를 관리하면 상태의 이유가 되는 다른 정보(duration)와 불일치가 발생할 수 있음

**Comment by kyukong** [comment]
현재는 2안, 이후에 사용자가 직접 그룹의 상태를 컨트롤하는 기능이 추가된다면 1안으로 바꾸면 좋을것 같습니다.

지금까지 구현된 상태로는 마감 기한이 끝났던가, 모집 인원이 다 찼던가 등의 데이터를 기반으로 상태가 바뀝니다. 상태를 업데이트 하는 것을 빠트린다면 실제 상태와 반영된 상태가 다른 경우가 발생할 수도 있을 것 같습니다. 

이후에 사용자가 직접 상태를 조절(모집 마감 등) 한다면 데이터 기반이 아닌 외부의 개입이 있기 때문에 별도의 상태 정보를 관리해야 할듯합니다.

**Reply by sinb57** [reply]
동의합니다!
